mantidproject/mantid
# Elwin tab presenter is no longer a QObject and View is passed to constructor.

### Description of work
This PR removes the Qt dependency of the presenter of the Elwin tab. The implementation follows very closely the layout set in #36478 , and more justification for the changes can be found therein. 
#### Summary of work




Part of  #36477  . 




#### Further detail of work

- This tab in particular had a tightly bound presenter/view, as the presenter additionally held a workspace dialog interface. This dialog has been moved to the view. This has required to refactor the functions calling the interface from the presenter to the view.
- Additionally, an unspotted bug which prevented the spectrum from being updated when the index was raised or lowered from the QSpinbox element has been fixed. 
### To test:
1. Open `Data Manipulation -> Elwin`
2. Open a test file `irs2673_graphite002_red.nxs` : either from file or adding as a workspace
3. Check all view options work as intended: checkbox, edits, graph widgets.
4. Run the algorithm with appropriate input parameters. 
5. Check output is valid and as expected. 








---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
